lib/net.c

```diff
@@ -1641,42 +1629,38 @@ init_net_stage1(void *arg)
 
 	calculate_net_config_para(net_conf);
 
-	if (net_conf->gatekeeper_pktmbuf_pool == NULL) {
-		net_conf->gatekeeper_pktmbuf_pool =
-			rte_calloc("mbuf_pool", net_conf->numa_nodes,
-				sizeof(struct rte_mempool *), 0);
-		if (net_conf->gatekeeper_pktmbuf_pool == NULL) {
-			G_LOG(ERR, "net: %s: out of memory\n", __func__);
-			return -1;
-		}
-	}
-
-	/* Initialize pktmbuf pool on each used NUMA node. */
-	for (i = 0; i < net_conf->numa_nodes; i++) {
+	/* Initialize pktmbuf pool on each used lcore. */
+	for (i = 0; i < RTE_MAX_LCORE; i++) {
 		char pool_name[64];
 		int ret;
 
-		if (!net_conf->numa_used[i] ||
+		if (!net_conf->lcores_used[i] ||
 				net_conf->gatekeeper_pktmbuf_pool[i] != NULL)
 			continue;
 
 		ret = snprintf(pool_name, sizeof(pool_name), "pktmbuf_pool_%u",
 			i);
 		RTE_VERIFY(ret > 0 && ret < (int)sizeof(pool_name));
 		net_conf->gatekeeper_pktmbuf_pool[i] =
-			rte_pktmbuf_pool_create(pool_name,
+			rte_mempool_create(pool_name,
```

Comment:
Since blocks will now create their own pools, it's better to encapsulate this call with a function.